Shared/Level/Level.gd

```diff
@@ -2,49 +2,63 @@ extends Spatial
 class_name Level
 
 
-# Scene to add  to the capture points for logic
+# Scene to add to the capture and spawn points for logic
+# defined in UI/Menus/StartMenu for client
+# 		 and GameRoom/GameRoom for server
 export var capture_point_scene: PackedScene
-var _capture_points =[]
+var _capture_points = []
+
+# spawn point visuals for client
+export var spawn_point_scene: PackedScene
 
 func _ready():
 	if not capture_point_scene:
 		Logger.error("No Capture Point Scene! Capture Points will not work")
 		return
-	else:
-		for point in $CapturePoints.get_children():
-			var new_scene = capture_point_scene.instance()
-			new_scene.global_transform = point.global_transform
-			_capture_points.append(new_scene)
-			add_child(new_scene)
-	
-
-func color_spawn_points(player_team_id):
+
+	for point in $CapturePoints.get_children():
+		var new_scene = capture_point_scene.instance()
+		_capture_points.append(new_scene)
+		point.add_child(new_scene)
+
+	if not spawn_point_scene:
+		Logger.error("No valid Spawn Point Scene! Spawn Points will not work")
+		return
+
+	# client only properties -> server does not need any visuals
+	var max_ghosts = Constants.get_value("ghosts", "max_amount")
 	for player in range(2):
-		var spawns = "Player" + str(player + 1) + "Spawns"
-		if has_node(spawns):
-			var index = 0
-			for spawn in get_node(spawns).get_children():
-				var color_scheme = "player_" if player_team_id == player else "enemy_"
-				var wall_index = Constants.get_value("ghosts","wall_placing_timeline_index")
-				var accent_type = "primary" if wall_index != index else "secondary"
-				var color = Color(Constants.get_value("colors", color_scheme+accent_type+"_accent"))
-				spawn.get_node("MeshInstance").material_override.albedo_color = color
-				index += 1
+		var spawns = get_node("Player" + str(player + 1) + "Spawns")
+		for spawn in spawns.get_children():
+			var new_scene = spawn_point_scene.instance()
+			# TODO: could check if configured scene is valid
+			#if new_scene.has_method("set_type"):
+
+			# set action type depending on point index
+			new_scene.set_type(spawn.get_index() % max_ghosts)
+			# rotate depending on player index: 0 starts left, 1 right
+			new_scene.rotate_ui(player * 180)
+			# only activate for current player in prep phase
+
+			spawn.add_child(new_scene)
+
 
 func reset():
 	for capture_point in _capture_points:
 		capture_point.reset()
+
 	toggle_capture_points(false)
 	toggle_spawn_points(true)
 
+
 func get_spawn_points(team_id):
 	var node_name = "Player" + str(team_id + 1) + "Spawns"
 	if has_node(node_name):
-		var spawn_positions = []
-		for position in get_node(node_name).get_children():
-			spawn_positions.append(position.global_transform.origin)
-
-		return spawn_positions
+		# EDIT: return full node instead of only global_transform vector
```

Comment:
WIP comment?

Reply:
removed, was just in for tracing reasons if bugs occur during this pr